### DISCUSSION TITLE
Reducing/removing timings from server replays

### DISCUSSION THREAD

**Discussion by jared-heyautofill** [mitmproxy/mitmproxy#5934] [Q&A]
I have an app that I'm working on that reads some content from 3rd party websites. In testing, I've saved a page load of this website using mitmdump. I then use mitmproxy to respond to requests during the test run, so I don't continuously hit 3rd party servers during testing runs.

During these test runs, mitmproxy reproduces the timings of the initial recording. This has significant negative impacts on the time to complete the test. I've looked through all of the options of mitmproxy and don't see an option to ignore/reduce timings.

Is this possible? Could I use mitmproxy in reverse proxy mode to do this (note that there are many domains that are hit due to various resource caching strategies, 3rd party resources, etc.?

**Comment by mhils** [comment]
> During these test runs, mitmproxy reproduces the timings of the initial recording.

That shouldn't be the case, mitmproxy should reply without delay. Here's the relevant code if that helps: https://github.com/mitmproxy/mitmproxy/blob/main/mitmproxy/addons/serverplayback.py#L232 - there are no delays in the logic at all.

One thing you might want to look into is setting `connection_strategy` to lazy.
